src/view/treeNodes/fileChangeNode.ts

```diff
@@ -14,6 +14,7 @@ import { FolderRepositoryManager, SETTINGS_NAMESPACE } from '../../github/folder
 import { IResolvedPullRequestModel, PullRequestModel } from '../../github/pullRequestModel';
 import { FileChangeModel, GitFileChangeModel, InMemFileChangeModel, RemoteFileChangeModel } from '../fileChangeModel';
 import { DecorationProvider } from '../treeDecorationProvider';
+import { DirectoryTreeNode } from './directoryTreeNode';
```

Comment:
Now we have a dependency cycle where `FileChangeNode` depends on `DirectoryTreeNode` and vice versa. It looks like this dependency exists for `updateParentCheckbox`. I see that you've already added `updateCheckbox` to `TreeNode`.  Can `updateParentCheckbox` be added to `TreeNode` as well?